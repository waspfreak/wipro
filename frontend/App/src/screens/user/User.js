import React from 'react';
import { Button, Text, ScrollView } from 'react-native';
import styles from './styles';
import { useGlobalStore } from '../../stores/globalStore';

export const User = () => {
  const [oneTaskStore, dispatch] = useGlobalStore();
  const { userStatus, signOut } = oneTaskStore;

  console.log('oneTaskStore', oneTaskStore);
  return (
    <ScrollView contentContainerStyle={styles.container}>
      <Text>The User screen aka 'My stuff' lives here</Text>
      <Button
        title="Test global state"
        onPress={() =>
          dispatch({ type: 'SIGN_IN', payload: { userStatus: false } })
        }
      />
      <Button title="Sign Out" tertiary onPress={signOut} />
      <Text>{JSON.stringify({ userStatus })}</Text>
    </ScrollView>
  );
};

export default User;
