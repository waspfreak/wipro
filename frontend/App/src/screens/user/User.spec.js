import React from 'react';
import { render } from '@testing-library/react-native';
import User from '../user/User';

describe('User tests', () => {
  it('should match User snapshot', () => {
    const { baseElement } = render(<User />);
    expect(baseElement).toMatchSnapshot();
  });
});
