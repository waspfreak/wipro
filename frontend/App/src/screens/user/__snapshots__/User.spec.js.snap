// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User tests should match User snapshot 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <ScrollView
      contentContainerStyle={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flex": 1,
          "justifyContent": "center",
          "paddingBottom": 50,
          "paddingTop": 15,
        }
      }
    >
      <Text>
        The User screen aka 'My stuff' lives here
      </Text>
      <Button
        title="Test global state"
      />
      <Button
        tertiary={true}
        title="Sign Out"
      />
      <Text>
        {}
      </Text>
    </ScrollView>
  </View>
</View>
`;
